java-desktop/src/main/java/com/couchbase/todo/TodoApp.java

```diff
@@ -35,34 +45,35 @@ public void stop() throws Exception {
         super.stop();
     }
 
-    public static void gotoLoginScreen(Stage stage) {
-        try {
-            FXMLLoader loader = new FXMLLoader(TodoApp.class.getResource("/scene/Login.fxml"));
+    public static void setScene(Stage stage, String fxmlFile) {
+        FXMLLoader loader = new FXMLLoader(TodoApp.class.getResource(fxmlFile));
+
+        if (fxmlFile.equals(MAIN_FXML)) {
+            loader.setController(new MainController(stage));
+        }
+        else if (fxmlFile.equals(LOGIN_FXML)) {
             loader.setController(new LoginController(stage));
-            Parent root = loader.load();
-            Scene scene = new Scene(root);
-            stage.setScene(scene);
         }
-        catch (IOException e) {
-            throw new IllegalStateException(e);
+        else if (fxmlFile.equals(CONFIG_FXML)) {
+            loader.setController(new ConfigController(stage));
         }
+
+        Parent root;
+        try { root = loader.load(); }
+        catch (IOException e) { throw new RuntimeException("IOException when set scene to : " + fxmlFile, e); }
+
+        Scene scene = new Scene(root);
+        stage.setScene(scene);
     }
 
-    public static void gotoMainScreen(Stage stage) {
-        try {
-            FXMLLoader loader = new FXMLLoader(TodoApp.class.getResource("/scene/main.fxml"));
-            loader.setController(new MainController(stage));
-            Parent root = loader.load();
-            Scene scene = new Scene(root);
-            stage.setScene(scene);
-        }
-        catch (IOException e) {
-            throw new IllegalStateException(e);
-        }
+    public static synchronized Config getConfig() {
+        if (config == null) { config = new Config(Config.builder()); }
+        return config;
     }
 
-    public static void main(String[] args) {
-        launch(args);
+    public static synchronized void setConfig(Config newConfig) {
```

Comment:
Yeah... in your crusade against nulls, though, I don't think you just just ignore a call to set the config null.  I think you should probably throw an NPE, right?